src/highlight_provider.ts

```diff
@@ -452,7 +267,7 @@ export class HighlightProvider implements Disposable {
 
         const gridHl = this.highlights.get(grid);
         if (gridHl) {
-            gridHl.forEach((rowHighlights, row) => {
```

Comment:
Github won't let me put this comment on the line I want, but one question I have for those more familiar with this code: is the highlight created by `createColVirtTextOptions` strictly for virtual text? I only ask because on line 467, the check tests for either `colHighlights.length > 1` or `colHighlights[0].virtText`, which implies to me there may be some other case other than virtual text where this gets triggered. If there is, what is it?

Reply:
@xiyaowong Sorry for the ping - if you have a moment and wouldn't mind weighing in here, that would be appreciated. The blame says this came from your refactor, so I figure if anyone would know, it's you :)

Reply:
The virtText and the original text of cell is the same.
- `colHighlights.length > 1` indicates virtual text on multi-width characters, which include 2 cells.
- `colHighlights[0].virtText` indicates virtual text on single-width characters, and this value is used to distinguish normal highlights from virtual text.
The above two conditions essentially represent "isVirtText".

Reply:
> is the highlight created by `createColVirtTextOptions` strictly for virtual text?

Yes.

> the check tests for either `colHighlights.length > 1` or `colHighlights[0].virtText`

The virtText and the original text of cell is the same.

- `colHighlights.length > 1` indicates virtual text on multi-width characters, which include 2 cells.
- `colHighlights[0].virtText` indicates virtual text on single-width characters, and this value is used to distinguish normal highlights from virtual text.

The above two conditions essentially represent "isVirtText".

Reply:
Off the top of my head, this might also support decorators that don't need to be virtual text but that workaround vscode limitations, like text/fake cursor after EOL.

Reply:
Do you know of a situation where text after EOL would be used? Just trying to take in all of this

Reply:
Just consider that any text that is inconsistent with the original text is treated as virtual text.

Reply:
> Do you know of a situation where text after EOL would be used? Just trying to take in all of this

Onemore visual cursor, where the fake visual cursor is placed after eol. 

Leap.nvim, if the label is on the last character.